pkg/api/db.go

```diff
@@ -213,3 +215,432 @@ func DeleteVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) error {
 	}
 	return nil
 }
+
+func CreateVolumeGroupDBEntry(ctx *c.Context, in *model.VolumeGroupSpec) (*model.VolumeGroupSpec, error) {
+	if in.Id == "" {
+		in.Id = uuid.NewV4().String()
+	}
+	if in.AvailabilityZone == "" {
+		log.Warning("Use default availability zone when user doesn't specify availabilityZone.")
+		in.AvailabilityZone = "default"
+	}
+
+	group := &model.VolumeGroupSpec{
+		BaseModel: &model.BaseModel{
+			Id: in.Id,
+		},
+		UserId:           ctx.UserId,
+		Name:             in.Name,
+		Description:      in.Description,
+		AvailabilityZone: in.AvailabilityZone,
+		Status:           model.VOLUMEGROUP_CREATING,
+	}
+	result, err := db.C.CreateVolumeGroup(ctx, group)
+	if err != nil {
+		log.Error("When add volume to db:", err)
+		return nil, err
+	}
+	// TODO:Rpc call to create group.
+	// Create volume group request is sent to the Dock. Dock will update volume status to "available"
+	// after volume group creation is completed.
+	controller.Brain.CreateVolumeGroup(ctx, group)
```

Comment:
It is quite odd if you put calling controller rpc methods in this file. IMO `api/db.go` represents the connection between api-server and etcd, you can't merge the logic between api-server and controller in this file.

Reply:
I may change the name of this file. In my opinion, the rpc call can not be implemented in api file either. In cinder, rpc call and db operation are implemented in the same function.

Reply:
rpc call is just a sentence in the function.